### PR TITLE
[ONNX] Refactor ONNX fixup for Loop and If

### PR DESCRIPTION
**Pull request by BowenBao**
* move both under new file `fixup_onnx_controlflow`
* move the fixup to where the ONNX loop/if node is created, as oppose to running the fixup as postpass. This will help with enable onnx shape inference later.
* move `fuseSequenceSplitConcat` to `Peephole`.

### CODE REVIEW

File: torch/csrc/jit/passes/onnx/peephole.cpp [hunk 1]
```diff
@@ -948,6 +948,48 @@ static void fuseLogSoftmaxNllLoss(Block* b) {
   }
 }
 
+static void fuseSequenceSplitConcat(Block* b) {
```
Comment (neginraoof) [inline torch/csrc/jit/passes/onnx/peephole.cpp:951]: Is this fuse needed for export of loops?
Can you add comments to describe it?

**Review by neginraoof — COMMENTED** [review]
Looks good. Thanks for the update. Left a comment.

File: torch/csrc/jit/passes/onnx/peephole.cpp (hunk 1 above)
Comment (BowenBao) [reply torch/csrc/jit/passes/onnx/peephole.cpp]: yes, export of loops sometimes creates (split -> concat) pair which is actually no-op. This optimization removes such cases if possible. Updated the name to `removeSequenceSplitConcat`, and added comments.

**Review by neginraoof — APPROVED** [review]
LGTM. Thanks!
I don't think failures are related. Please rebase.

**Review by bzinodev — APPROVED** [review]

**Comment by BowenBao** [comment]
rebased to resolve conflict

**Pull request closed without merging**